warehouse/query-core/src/main/java/datawave/query/util/DateIndexHelper.java

```diff
@@ -170,39 +173,93 @@ public DateIndexHelper initialize(AccumuloClient client, String dateIndexTableNa
 
     public static class DateTypeDescription {
         final Set<String> fields = new HashSet<>();
-        final String[] dateRange = new String[2];
+        private String start;
+        private String end;
+
+        public void addField(String field) {
+            this.fields.add(field);
+        }
 
         public Set<String> getFields() {
             return fields;
         }
 
         public String[] getDateRange() {
-            return dateRange;
+            return new String[] {start, end};
+        }
+
+        /**
+         * Method used to set the default values for the start and end date if none were found
+         *
+         * @param startDate
+         *            the start date
+         * @param endDate
+         *            the end date
+         */
+        public void setStartEndDate(Date startDate, Date endDate) {
```

Comment:
I did not see a check in `DateIndexHelper.getDateTypeDescription()` to assert that the arguments `begin` and `end` are not null. I suggest adding checks here to only update start/end if startDate/endDate are not null. Additionally, the method name could make someone think they are always setting the start and end date if they don't read the documentation. Perhaps a name like `replaceNullDateRangeEndpoints()` or something like that?